ansible/roles/images/tasks/check-or-pull-images.yaml

```diff
@@ -0,0 +1,21 @@
+---
+# If image is present, its name is written to stdout. Return code is 0.
+#   > ctr --namespace k8s.io images check -q name==docker.io/mesosphere/pause-alpine:3.2; echo $?
+#   docker.io/mesosphere/pause-alpine:3.2
+#   0
+#
+# If image is not present, or is present, but some layer is missing, its name is not written to stdout. Return code is 0.
+#   > ctr --namespace k8s.io images check -q name==docker.io/mesosphere/pause-alpine:999; echo $?
+#   0
+- name: check existing image to ensure all content is available locally  # noqa no-changed-when
+  command: ctr --namespace k8s.io images check name=={{ image_name }}
```

Comment:
Yay tests https://teamcity.mesosphere.io/buildConfiguration/ClosedSource_KonvoyImageBuilder_E2e/3310128?buildTab=log&focusLine=18000&linesState=409

KIB configures `crictl` with the correct socket but looks `ctr` doesn't get those configs, we need something similar to:
https://github.com/mesosphere/konvoy-image-builder/blob/main/ansible/roles/config/templates/crictl.yaml

Reply:
An alternative is to keep using `crictls`

```
[root@ip-10-0-71-7 bin]# crictl inspecti -q k8s.gcr.io/pause:3.4.1
{
  "status": {
    "id": "sha256:0f8457a4c2ecaceac160805013dc3c61c63a1ff3dee74a473a36249a748e0253",
    "repoTags": [
      "k8s.gcr.io/pause:3.4.1"
    ],
    "repoDigests": [
      "k8s.gcr.io/pause@sha256:6c3835cab3980f11b83277305d0d736051c32b17606f5ec59f1dda67c9ba3810"
    ],
    "size": "301268",
    "uid": null,
    "username": "",
    "spec": null
  }
}
[root@ip-10-0-71-7 bin]# crictl inspecti -q k8s.gcr.io/pause:3.4.2
FATA[0000] no such image "k8s.gcr.io/pause:3.4.2" present
```

Reply:
:raised_hands: Thanks tests! Will fix now.

I am surprised, because I tested the `provision` command against an EC2 machine that uses an AMI (ami-06956df2bea3220df, "konvoy-ami-centos-7-1.20.13-1639483726") that, AFAIK, is created by konvoy-image-builder.

So I expected the EC2 instance to accurately represent the state a machine is in just after the `build` command runs.

Reply:
Looked at the test results; this appears to be a problem **only** on flatcar. Correct?

Reply:
I don't see how we can import an image from a tarball with `crictl`.

Reply:
Yes only for Flatcar.

Oh good point, need to use `ctr.`